lib/pure/parseutils.nim

```diff
@@ -592,6 +592,71 @@ proc parseFloat*(s: openArray[char], number: var float): int {.
   if result != 0:
     number = bf
 
+func toLowerAscii(c: char): char =
+  if c in {'A'..'Z'}: char(uint8(c) xor 0b0010_0000'u8) else: c
+
+func parseSize*(s: openArray[char], size: var int64, alwaysBin=false): int =
+  ## Parse a size qualified by binary or metric units into `size`.  This format
+  ## is often called "human readable".  Result is the number of processed chars
+  ## or 0 on parse errors and size is rounded to the nearest integer.  Trailing
+  ## garbage like "/s" in "1k/s" is allowed and detected by `result < s.len`.
+  ##
+  ## To simplify use, following non-rare wild conventions, and since fractional
+  ## information is so rare, unit  matching is case-insensitive but for the 'i'
+  ## distinguishing binary-metric from metric which cannot be 'I'.
+  ##
+  ## An optional trailing 'B|b' is ignored but processed.  I.e., you must still
+  ## know if units are bytes | bits or infer this fact via the case of s[^1] (if
+  ## users can even be relied upon to use 'B' for byte and 'b' for bit or have
+  ## that be s[^1]).  Trailing text stops the parse, as usual.
+  ##
+  ## If `alwaysBin==true` then scales are always binary-metric, but e.g. "KiB"
+  ## is still accepted for clarity.  If the value would exceed the range of
+  ## `int64`, `size` is set to `int64.high`.  Supported metric prefix chars
```

Comment:
```suggestion
  ## `int64`, then `size` is saturated. Supported metric prefix chars
```

Reply:
Well, I suppose the same module does have `parseSaturatedNatural`, but I was thinking being more specific might be helpful.  Maybe first occurrence "saturated to `int64.high`" instead of "set to" and second occurrence only "z & y saturate"?